### ISSUE TITLE
Multi-Session nicks: sometimes the wrong client is removed

### ISSUE THREAD

**Issue by ge0rg** [matrix-org/matrix-bifrost#93]
I'm joined from to a bridged Matrix room, using two different clients: georg@yax.im/poezio and georg@yax.im/badconnection

badconnection is having network outages all the time, leaving, joining, reconnecting.

poezio is stable and reliable. Also, poezio is periodically doing a [self-ping](https://xmpp.org/extensions/xep-0410.html#performingselfping).

Suddenly, poezio is getting back an error code:

```xml
SEND:
<iq id="cb0186cb61824d1d9f5e8ba333ccc0cc" type="get" to="#bifrost#half-shot.uk@half-shot.uk/Ge0rG">
  <ping xmlns="urn:xmpp:ping" />
</iq>

RECV:
<iq id="cb0186cb61824d1d9f5e8ba333ccc0cc" type="error" to="georg@yax.im/poezio-IS8H" from="#bifrost#half-shot.uk@half-shot.uk/Ge0rG" xml:lang="en">
  <error type="cancel" by="#bifrost#half-shot.uk@half-shot.uk">
    <not-acceptable xmlns="urn:ietf:params:xml:ns:xmpp-stanzas" />
  </error>
</iq>
```

This error code makes poezio think that it's disconnected, so it forcefully leaves and rejoins the room.

**Comment by ge0rg** [comment]
I just had another problem of either this or something related to your #107 change:

I restarted my mobile client (yaxim), and my desktop client received a presence-unavailable:

```xml
2020-02-14 12:09:39.475 RECV:
<presence from="#matrix#matrix.org@matrix.org/Ge0rG" id="32a1944869f74f84b281707617d9b473" to="georg@yax.im/poezio-IS8H" type="unavailable">
  <x xmlns="http://jabber.org/protocol/muc#user">
    <item affiliation="member" jid="georg@yax.im/yaxim" role="none" />
  </x>
</presence>
```

yaxim will send a presence-unavailable to a MUC before joining, because old MUC implementations  failed to  send a full synchronization to a client if they thought the client still was joined.

You need to ignore presence-unavailable from a resource that is not joined to the MUC.

**Comment by ge0rg** [comment]
I've had a deeper look into the issue and created the following patch:

https://op-co.de/tmp/0001-Only-reflect-MUC-leave-when-last-XMPP-device-leaves-.patch

**Comment by Half-Shot** [comment]
This was fixed by #140
